presto-jdbc/src/main/java/io/prestosql/jdbc/PrestoDriverUri.java

```diff
@@ -359,16 +389,23 @@ private static Properties mergeConnectionProperties(URI uri, Properties driverPr
         }
 
         Properties result = new Properties();
-        setProperties(result, defaults);
         setProperties(result, urlProperties);
         setProperties(result, suppliedProperties);
         return result;
     }
 
+    private static void setDefaultPropertiesIfNotProvided(Properties properties)
+    {
+        Map<String, String> defaults = ConnectionProperties.getDefaults();
+        setProperties(properties, defaults);
+    }
+
     private static void setProperties(Properties properties, Map<String, String> values)
     {
         for (Entry<String, String> entry : values.entrySet()) {
-            properties.setProperty(entry.getKey(), entry.getValue());
+            if (!properties.containsKey(entry.getKey())) {
```

Comment:
This affects `mergeProvidedProperties` and things that were overwritten there will be no longer overwritten.